runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/engine/HttpClientEngineConfig.kt

```diff
@@ -4,7 +4,123 @@
  */
 package aws.smithy.kotlin.runtime.http.engine
 
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
+// See https://github.com/aws/aws-sdk-java-v2/blob/master/http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java
+// for all the options the Java v2 SDK supports
+
 /**
  * Common configuration options to be interpreted by an underlying engine
+ *
+ * NOTE: Not all engines will support every option! Engines *SHOULD* log a warning when given a configuration
+ * option they don't understand/support
  */
-class HttpClientEngineConfig
+@OptIn(ExperimentalTime::class)
+open class HttpClientEngineConfig constructor(builder: Builder) {
```

Comment:
**Question**: Do we see the members of this class growing over time? If so, I think a layer of organization might be in order. Otherwise this may turn into a difficult class to manage by users.